jobs/transforms/auth_audits.py

```diff
@@ -0,0 +1,66 @@
+# coding: utf-8
+#
+# Copyright 2021 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Beam DoFns and PTransforms to provide validation of auth models."""
+
+from __future__ import absolute_import  # pylint: disable=import-only-modules
+from __future__ import unicode_literals  # pylint: disable=import-only-modules
+
+from core.platform import models
+import feconf
+from jobs.decorators import audit_decorators
+from jobs.transforms import base_model_audits
+
+(auth_models, user_models) = (
+    models.Registry.import_models([models.NAMES.auth, models.NAMES.user]))
+
+
+@audit_decorators.AuditsExisting(auth_models.FirebaseSeedModel)
+class ValidateFirebaseSeedModelId(base_model_audits.ValidateBaseModelId):
+    """Overrides regex to match the single valid FirebaseSeedModel ID."""
+
+    def __init__(self):
+        super(ValidateFirebaseSeedModelId, self).__init__()
+        self._pattern = auth_models.ONLY_FIREBASE_SEED_MODEL_ID
+
+
+@audit_decorators.AuditsExisting(auth_models.UserIdByFirebaseAuthIdModel)
+class ValidateUserIdByFirebaseAuthIdModelId(
+        base_model_audits.ValidateBaseModelId):
+    """Overrides regex to match the Firebase account ID pattern."""
```

Comment:
Hm, do you think it would be better to (instead of having a "wrong" value that gets overridden) _require_ each subclass to explicitly specify a `pattern` property? It seems reasonable for IDs of different model classes to have independent generation logic.

Reply:
I don't think so, since some models have special logic that determines the regexp by inspecting the model (for example, `TaskEntry.id == {TaskEntry.task_type}.{TaskEntry.target_type}.{TaskEntry.target_id}. (etc.)`).

Forcing an override of `pattern` in init is too early to create that pattern, so I don't want to enforce that subclasses use it.

Reply:
OK, sg.